clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -673,6 +675,61 @@ public void testWakeupWithFetchDataAvailable() {
         ConsumerRecords<String, String> records = consumer.poll(0);
         assertEquals(5, records.count());
     }
+    
+    @Test
+    public void testCommitSyncTerminatesIfInterrupted() throws Exception {
+        int rebalanceTimeoutMs = 60000;
+        int sessionTimeoutMs = 30000;
+        int heartbeatIntervalMs = 3000;
+
+        final Time time = new MockTime();
+        Cluster cluster = TestUtils.singletonCluster(topic, 1);
+        final Node node = cluster.nodes().get(0);
+
+        Metadata metadata = new Metadata(0, Long.MAX_VALUE);
+        metadata.update(cluster, time.milliseconds());
+
+        final MockClient client = new MockClient(time, metadata);
+        client.setNode(node);
+        final PartitionAssignor assignor = new RoundRobinAssignor();
+
+        final KafkaConsumer<String, String> consumer = newConsumer(time, client, metadata, assignor,
+                rebalanceTimeoutMs, sessionTimeoutMs, heartbeatIntervalMs, false, 0);
+        
+        Runnable consumerRunner = new Runnable() {
+            @Override
+            public void run() {
+                consumer.subscribe(Arrays.asList(topic), getConsumerRebalanceListener(consumer));
+                prepareRebalance(client, node, assignor, Arrays.asList(tp0), null);
+
+                consumer.poll(0);
+
+                // respond to the outstanding fetch so that we have data available on the next poll
+                client.respondFrom(fetchResponse(tp0, 0, 5), node);
+                client.poll(0, time.milliseconds());
+                
+                //Get the messages
+                consumer.poll(0);
+                
+                // Interrupt the thread and call commitSync
+                Thread.currentThread().interrupt();
```

Comment:
I see. Alternatively, you could verify the interrupt in the call to `poll()` instead of `commitSync()`, since it already has a timeout.